src/route-planning/cargoUnit.js

```diff
@@ -62,14 +71,36 @@ class CargoUnit extends OrbitingGameObject {
             linVel.normalize();
 
             this.timeInTransit += deltaTime;
-            this.distanceTraveled += this.lastVelocity.length() * deltaTime;            
-            
+            this.distanceTraveled += this.lastVelocity.length() * deltaTime;           
             this.rotation = Vector3.Cross(this.mesh.up, linVel);
+
+            this.captureRouteData();
+            this.encounterManager.update(deltaTime);
             this.physicsImpostor.applyImpulse(this.currentGravity.scale(deltaTime), this.mesh.getAbsolutePosition());
             this.currentGravity = Vector3.Zero();
         }
     }
 
+    captureRouteData() {
+        let node = new TransformNode("cargoNode", this.scene, true);
```

Comment:
can be const.

Also, can this be re-used in any way? instead of creating a new one?

Another issue - those transform nodes should be disposed at a certain point, otherwise they will stay in the scene forever.

Reply:
Hmm -- I was wondering about the number of Nodes created myself. The Nodes will be used to construct a tube, path3D, etc.. that the camera will follow, but I wanted to be able to link encounters to them easily via the GUI.Control.linkToMesh function. I'll add a loop through the route list to dispose the nodes when the scene is reset; they'll be cleaned up with the scene when it's disposed though right, or do I need to manually account for that as well?

Reply:
Something I was thinking of doing was to lower the sample rate so that there are fewer nodes created -- I don't think it's entirely necessary to capture the flight telemetry at 60Hz! 🕐

Reply:
as long as you dispose those meshes it won't really hurt performance. Also, check if it is possible to get away with just a transformation (matrix?) instead of a whole node to represent the point. might save a few cpu ticks :-)